packages/mask/src/plugins/Referral/helpers/buffer.ts

```diff
@@ -0,0 +1,78 @@
+import { BigNumber } from '@ethersproject/bignumber'
+import type { PrefixedHexString } from 'ethereumjs-util'
+
+// 32-byte buffer for hashes and 32-byte encoded strings
+export interface Buffer32 extends Buffer {}
+// 24-byte buffer for chain addresses and token definitions
+export interface Buffer24 extends Buffer {}
+// 24-byte buffer for chain addresses and token definitions
+export interface Buffer20 extends Buffer {}
```

Comment:
Prefer `Uint8Array` than `Buffer`. And please read this article https://betterprogramming.pub/nominal-typescript-eee36e9432d2 if you want a nominal type to distinguish those buffer types.

Reply:
I encourage that, but since we will never get rid of `buffer`, it's also ok to preserve that. Please make sure you import from the `buffer` module instead of using the global buffer.

Reply:
We are doing to use Uint8Array in the other our services as well.
So, for now, I imported  Buffer from the module and add "TODO" to use Uint8Array in the next release.

Reply:
Done it.